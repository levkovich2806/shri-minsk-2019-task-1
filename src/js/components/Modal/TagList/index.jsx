import React from "react";
import styles from "./index.module.scss";

function TagList(props) {
  const { tags } = props;

  return (
    <div className={styles.taglist}>
      <span className={styles.taglist__title}>Тэги: </span>
      <select
        multiple
        size="3"
        className={styles.taglist__list}
        onChange={props.onChange}
      >
        {tags.map(({ id, tag }) => (
          <option key={tag} value={id} className={styles.taglist__list_item}>
            {tag}
          </option>
        ))}
      </select>
    </div>
  );
}

export default TagList;
