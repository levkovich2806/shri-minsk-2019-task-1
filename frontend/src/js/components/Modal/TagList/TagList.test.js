import React from 'react';
import { shallow, mount } from 'enzyme';

import TagList from './index';

describe('TagList', () => {

  it('Should match snapshot TagList', () => {

    const tags = [
      {
        "id": 0,
        "tag": "покупки"
      },
      {
        "id": 1,
        "tag": "Работа"
      },
      {
        "id": 2,
        "tag": "ШРИ"
      },
      {
        "id": 3,
        "tag": "коты"
      },
      {
        "id": 4,
        "tag": "мемы"
      },
      {
        "id": 5,
        "tag": "универ"
      },
      {
        "id": 6,
        "tag": "домашнее задание"
      },
      {
        "id": 7,
        "tag": "учеба"
      },
      {
        "id": 8,
        "tag": "Neque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit"
      }
    ];
    const checkedTags = [2, 5];

    const wrapper = shallow(<TagList tags={tags} checkedTags={checkedTags} />);
    expect(wrapper).toMatchSnapshot();
  });
});
