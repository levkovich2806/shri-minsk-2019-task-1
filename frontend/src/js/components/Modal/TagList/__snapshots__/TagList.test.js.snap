// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagList Should match snapshot TagList 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TagList
    checkedTags={
      Array [
        2,
        5,
      ]
    }
    tags={
      Array [
        Object {
          "id": 0,
          "tag": "покупки",
        },
        Object {
          "id": 1,
          "tag": "Работа",
        },
        Object {
          "id": 2,
          "tag": "ШРИ",
        },
        Object {
          "id": 3,
          "tag": "коты",
        },
        Object {
          "id": 4,
          "tag": "мемы",
        },
        Object {
          "id": 5,
          "tag": "универ",
        },
        Object {
          "id": 6,
          "tag": "домашнее задание",
        },
        Object {
          "id": 7,
          "tag": "учеба",
        },
        Object {
          "id": 8,
          "tag": "Neque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="taglist__title"
        >
          Тэги: 
        </span>,
        <select
          className="taglist__list"
          multiple={true}
          size="3"
          value={
            Array [
              2,
              5,
            ]
          }
        >
          <option
            className="taglist__list_item"
            value={0}
          >
            покупки
          </option>
          <option
            className="taglist__list_item"
            value={1}
          >
            Работа
          </option>
          <option
            className="taglist__list_item"
            value={2}
          >
            ШРИ
          </option>
          <option
            className="taglist__list_item"
            value={3}
          >
            коты
          </option>
          <option
            className="taglist__list_item"
            value={4}
          >
            мемы
          </option>
          <option
            className="taglist__list_item"
            value={5}
          >
            универ
          </option>
          <option
            className="taglist__list_item"
            value={6}
          >
            домашнее задание
          </option>
          <option
            className="taglist__list_item"
            value={7}
          >
            учеба
          </option>
          <option
            className="taglist__list_item"
            value={8}
          >
            Neque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit
          </option>
        </select>,
      ],
      "className": "taglist",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Тэги: ",
          "className": "taglist__title",
        },
        "ref": null,
        "rendered": "Тэги: ",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option
              className="taglist__list_item"
              value={0}
            >
              покупки
            </option>,
            <option
              className="taglist__list_item"
              value={1}
            >
              Работа
            </option>,
            <option
              className="taglist__list_item"
              value={2}
            >
              ШРИ
            </option>,
            <option
              className="taglist__list_item"
              value={3}
            >
              коты
            </option>,
            <option
              className="taglist__list_item"
              value={4}
            >
              мемы
            </option>,
            <option
              className="taglist__list_item"
              value={5}
            >
              универ
            </option>,
            <option
              className="taglist__list_item"
              value={6}
            >
              домашнее задание
            </option>,
            <option
              className="taglist__list_item"
              value={7}
            >
              учеба
            </option>,
            <option
              className="taglist__list_item"
              value={8}
            >
              Neque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit
            </option>,
          ],
          "className": "taglist__list",
          "multiple": true,
          "onChange": undefined,
          "size": "3",
          "value": Array [
            2,
            5,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "покупки",
            "nodeType": "host",
            "props": Object {
              "children": "покупки",
              "className": "taglist__list_item",
              "value": 0,
            },
            "ref": null,
            "rendered": "покупки",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "Работа",
            "nodeType": "host",
            "props": Object {
              "children": "Работа",
              "className": "taglist__list_item",
              "value": 1,
            },
            "ref": null,
            "rendered": "Работа",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "ШРИ",
            "nodeType": "host",
            "props": Object {
              "children": "ШРИ",
              "className": "taglist__list_item",
              "value": 2,
            },
            "ref": null,
            "rendered": "ШРИ",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "коты",
            "nodeType": "host",
            "props": Object {
              "children": "коты",
              "className": "taglist__list_item",
              "value": 3,
            },
            "ref": null,
            "rendered": "коты",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "мемы",
            "nodeType": "host",
            "props": Object {
              "children": "мемы",
              "className": "taglist__list_item",
              "value": 4,
            },
            "ref": null,
            "rendered": "мемы",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "универ",
            "nodeType": "host",
            "props": Object {
              "children": "универ",
              "className": "taglist__list_item",
              "value": 5,
            },
            "ref": null,
            "rendered": "универ",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "домашнее задание",
            "nodeType": "host",
            "props": Object {
              "children": "домашнее задание",
              "className": "taglist__list_item",
              "value": 6,
            },
            "ref": null,
            "rendered": "домашнее задание",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "учеба",
            "nodeType": "host",
            "props": Object {
              "children": "учеба",
              "className": "taglist__list_item",
              "value": 7,
            },
            "ref": null,
            "rendered": "учеба",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "Neque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit",
            "nodeType": "host",
            "props": Object {
              "children": "Neque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit",
              "className": "taglist__list_item",
              "value": 8,
            },
            "ref": null,
            "rendered": "Neque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit",
            "type": "option",
          },
        ],
        "type": "select",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="taglist__title"
          >
            Тэги: 
          </span>,
          <select
            className="taglist__list"
            multiple={true}
            size="3"
            value={
              Array [
                2,
                5,
              ]
            }
          >
            <option
              className="taglist__list_item"
              value={0}
            >
              покупки
            </option>
            <option
              className="taglist__list_item"
              value={1}
            >
              Работа
            </option>
            <option
              className="taglist__list_item"
              value={2}
            >
              ШРИ
            </option>
            <option
              className="taglist__list_item"
              value={3}
            >
              коты
            </option>
            <option
              className="taglist__list_item"
              value={4}
            >
              мемы
            </option>
            <option
              className="taglist__list_item"
              value={5}
            >
              универ
            </option>
            <option
              className="taglist__list_item"
              value={6}
            >
              домашнее задание
            </option>
            <option
              className="taglist__list_item"
              value={7}
            >
              учеба
            </option>
            <option
              className="taglist__list_item"
              value={8}
            >
              Neque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit
            </option>
          </select>,
        ],
        "className": "taglist",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Тэги: ",
            "className": "taglist__title",
          },
          "ref": null,
          "rendered": "Тэги: ",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <option
                className="taglist__list_item"
                value={0}
              >
                покупки
              </option>,
              <option
                className="taglist__list_item"
                value={1}
              >
                Работа
              </option>,
              <option
                className="taglist__list_item"
                value={2}
              >
                ШРИ
              </option>,
              <option
                className="taglist__list_item"
                value={3}
              >
                коты
              </option>,
              <option
                className="taglist__list_item"
                value={4}
              >
                мемы
              </option>,
              <option
                className="taglist__list_item"
                value={5}
              >
                универ
              </option>,
              <option
                className="taglist__list_item"
                value={6}
              >
                домашнее задание
              </option>,
              <option
                className="taglist__list_item"
                value={7}
              >
                учеба
              </option>,
              <option
                className="taglist__list_item"
                value={8}
              >
                Neque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit
              </option>,
            ],
            "className": "taglist__list",
            "multiple": true,
            "onChange": undefined,
            "size": "3",
            "value": Array [
              2,
              5,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "покупки",
              "nodeType": "host",
              "props": Object {
                "children": "покупки",
                "className": "taglist__list_item",
                "value": 0,
              },
              "ref": null,
              "rendered": "покупки",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "Работа",
              "nodeType": "host",
              "props": Object {
                "children": "Работа",
                "className": "taglist__list_item",
                "value": 1,
              },
              "ref": null,
              "rendered": "Работа",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "ШРИ",
              "nodeType": "host",
              "props": Object {
                "children": "ШРИ",
                "className": "taglist__list_item",
                "value": 2,
              },
              "ref": null,
              "rendered": "ШРИ",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "коты",
              "nodeType": "host",
              "props": Object {
                "children": "коты",
                "className": "taglist__list_item",
                "value": 3,
              },
              "ref": null,
              "rendered": "коты",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "мемы",
              "nodeType": "host",
              "props": Object {
                "children": "мемы",
                "className": "taglist__list_item",
                "value": 4,
              },
              "ref": null,
              "rendered": "мемы",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "универ",
              "nodeType": "host",
              "props": Object {
                "children": "универ",
                "className": "taglist__list_item",
                "value": 5,
              },
              "ref": null,
              "rendered": "универ",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "домашнее задание",
              "nodeType": "host",
              "props": Object {
                "children": "домашнее задание",
                "className": "taglist__list_item",
                "value": 6,
              },
              "ref": null,
              "rendered": "домашнее задание",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "учеба",
              "nodeType": "host",
              "props": Object {
                "children": "учеба",
                "className": "taglist__list_item",
                "value": 7,
              },
              "ref": null,
              "rendered": "учеба",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "Neque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit",
              "nodeType": "host",
              "props": Object {
                "children": "Neque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit",
                "className": "taglist__list_item",
                "value": 8,
              },
              "ref": null,
              "rendered": "Neque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit",
              "type": "option",
            },
          ],
          "type": "select",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
